src/as/math/Plane.ts

```diff
@@ -21,22 +21,20 @@ export class Plane {
 	}
 
 	// set(normal: Vector3, constant: f32): Plane
-	// set: function(normal, constant) {
 	// 	this.normal.copy(normal)
 	// 	this.constant = constant
 
 	// 	return this
 	// },
 
-	setComponents(x: f32, y: f32, z: f32, w: f32): Plane {
+	setComponents(x: f32, y: f32, z: f32, w: f32): this {
```

Comment:
AFAICT `this` technically works, but I've been burned by using the type before. So my apologies.

Reply:
Huh, It's been working great for us,  so far at least.

The nice thing about it in TS (hopefully AS supports it), is that the `this` type extends to the subclasses ("F types"). So the return type is effectively automatically extended to be the subclass type if the method is called by an instance of a sub class.